src/google/protobuf/compiler/cpp/cpp_enum.cc

```diff
@@ -409,10 +409,10 @@ void EnumGenerator::GenerateMethods(int idx, io::Printer* printer) {
     std::string parent = ClassName(descriptor_->containing_type(), false);
     // Before C++17, we must define the static constants which were
     // declared in the header, to give the linker a place to put them.
-    // But pre-2015 MSVC++ insists that we not.
+    // But MSVC++ from pre-2015 to 2017 version 15.5 insists that we not.
```

Comment:
It sounds to me like the problem exists until 2015 *and* starting from 2017 version 15.5 and later, right? I guess MSVC fixed the bug in 2015 but reintroduced it in 2017. So maybe the comment should be more like: "But MSVC++ pre-2015 and post-2017 (version 15.5+) insists that we not."